### PR TITLE
PCQ-749: Add user consent cookie banner and cookie preferences saving feature.

### PR DESCRIPTION
**Pull request by JamesCollettCGI**
### JIRA link ###
https://tools.hmcts.net/jira/browse/PCQ-749

### Change description ###
PCQ-749: Add user consent cookie banner and cookie preferences saving feature.

Code changes :-
- app/assets/javascripts/cookie.js  --  Functions cookieManager.init , cookieBannerAccept , cookieBannerReject , cookieBannerSaved , preferenceFormSaved , cookiePreferencesUpdated
- app/views/includes/scripts.html  --  Removed reference to cookie_bundle.js (because it is now referenced via cookie-banner.html)
- app/views/includes/layout.html  --  Added reference to cookie-banner.html
- app/views/includes/cookie-banner.html (NEW)  --  Cookie Banner feature , references cookie_bundle.js
- app/steps/ui/static/cookies/template.html  --  User Preferences Saving feature
- app/resources/en/translation/common.json  --  Added 8 new strings
- app/resources/en/translation/static/cookies.json  --  Added 28 new strings
- app/resources/cy/translation/common.json  --  Added 8 new strings
- app/resources/cy/translation/static/cookies.json  --  Added 28 new strings

Cookie Banner feature implementation notes :- https://github.com/hmcts/cookie-manager#feature-cookie-banner
User Preferences Saving feature implementation notes :- https://github.com/hmcts/cookie-manager#feature-user-preferences-saving

See also screenshots attached to Jira ticket.

**Does this PR introduce a breaking change?**
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: app/resources/cy/translation/common.json [hunk 1]
```diff
@@ -21,6 +21,14 @@
   "months": "Ionawr, Chwefror, Mawrth, Ebrill, Mai, Mehefin, Gorffennaf, Awst, Medi, Hydref, Tachwedd, Rhagfyr",
   "supportLinks": "Cysylltiadau Cefnogi",
   "accessibility": "Datganiad Hygyrchedd",
+  "cookieBannerTitle": "TRANSLATE TO WELSH: Cookies on equality and diversity",
```
Comment (JamesCollettCGI) [inline app/resources/cy/translation/common.json:24]: Awaiting one String to be translated to Welsh ("Cookies on equality and diversity") , as able to find Welsh translations for all other new strings from other HMCTS projects.

File: app/assets/javascripts/cookie.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        'cookie-banner-reject-callback': cookieBannerReject,
+        'cookie-banner-accept-callback': cookieBannerAccept,
+        'cookie-banner-saved-callback': cookieBannerSaved,
+        'cookie-banner-auto-hide': false,
+        'cookie-manifest': [
+            // Additional GA cookies will need to be added here
+            {
+                'category-name': 'essential',
+                optional: false,
+                cookies: ['pcq-cookie-preferences', 'pcq-session', 'Idam.Session', 'seen_cookie_message'],
```
Comment (bradley-hmcts) [inline app/assets/javascripts/cookie.js:67]: add `_oauth2_proxy` to the essential cookie list

Comment (JamesCollettCGI) [reply app/assets/javascripts/cookie.js]: Thanks.   I've added _oauth2_proxy to essential cookie list.

**Comment by JamesCollettCGI** [comment]
Added _oauth2_proxy to essential cookie list.  
Added Welsh translation of Cookies on Equality and Diversity.

**Pull request merged**